### PR TITLE
Fix camera preview test

### PR DESCRIPTION
**Pull request by magangloff**
- Added back-camera to cirrus emulator
- Updated camera-preview 
- added camera-preview tests
- changed permission handling
- added permission tests

### CODE REVIEW

File: app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/CameraPreviewTest.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            case SurfaceRequest.Result.RESULT_SURFACE_ALREADY_PROVIDED:
+            case SurfaceRequest.Result.RESULT_WILL_NOT_PROVIDE_SURFACE:
+                Log.e("CameraPreviewTest", result.toString());
+                break;
+        }
+    }
+
+    @UiThreadTest
+    @Test
+    public void checkPreviewUseCase() throws InterruptedException, CameraInfoUnavailableException {
```
Comment (bgabriel998) [inline app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/CameraPreviewTest.java:121]: Add Simple comment

File: app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/MainActivityTest.java [hunk 2]
```diff
@@ -41,13 +51,47 @@ public void cleanUp(){
...
     public void intentIsFiredWhenUserClicksOnButton1() {
-        //Intents.init();
-
         onView(withId(R.id.button1)).perform(click());
         Intents.intended(IntentMatchers.hasComponent(Button1Activity.class.getName()));
-        //Intents.release();
     }
 
+    @Test
+    public void cameraPermissionRequested() throws NoSuchMethodException {
```
Comment (bgabriel998) [inline app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/MainActivityTest.java:59]: Add simple comment

File: app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/MainActivityTest.java [hunk 3]
```diff
@@ -41,13 +51,47 @@ public void cleanUp(){
...
+                e.printStackTrace();
+            }
+        });
+        onView(withText("Camera permission required!")).inRoot(RootMatchers.isDialog()).check(matches(isDisplayed()));
+        //Button1 is postive button of alertDialog
+        onView(withId(android.R.id.button1)).perform(click());
+    }
+
+    @Test
+    public void cameraPermissionGrantedAfterRequest() throws NoSuchMethodException {
```
Comment (bgabriel998) [inline app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/MainActivityTest.java:77]: Add simple comment

**Review by bgabriel998 — APPROVED** [review]
Just add few comments on for each test in the test file

File: app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/MainActivityTest.java (hunk 3 above)
Comment (magangloff) [reply app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/MainActivityTest.java:87]: Added comments

File: app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/MainActivityTest.java (hunk 2 above)
Comment (magangloff) [reply app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/MainActivityTest.java:65]: Added comments

File: app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/CameraPreviewTest.java (hunk 1 above)
Comment (magangloff) [reply app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/CameraPreviewTest.java]: Added comments

File: app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/GeonamesHandlerTest.java [hunk 4]
```diff
@@ -86,8 +86,8 @@ public void onResponseReceived(Object result) {
      */
     @Test
     public void testResultsQuantity(){
```
Comment (giommok) [inline app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/GeonamesHandlerTest.java:88]: Is this test necessary? It contains only comments

File: app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/GeonamesHandlerTest.java [hunk 5]
```diff
@@ -96,11 +96,11 @@ public void testResultsQuantity(){
      */
     @Test
     public void testResultType(){
```
Comment (giommok) [inline app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/GeonamesHandlerTest.java:98]: Is this test necessary? It contains only comments

**Review by giommok — APPROVED** [review]
Seems all fine.

**Pull request merged**